src/test/java/com/epam/aidial/cfg/functional/tests/AddonFunctionalTest.java

```diff
@@ -151,11 +152,41 @@ public void shouldThrowExceptionWhenRenameAddon() {
 
         IllegalArgumentException exception = Assertions.assertThrows(
                 IllegalArgumentException.class,
-                () -> addonFacade.updateAddon(addonDto.getName(), updatedAddon)
+                () -> addonFacade.updateAddon(addonDto.getName(), updatedAddon, "*")
         );
         Assertions.assertEquals("Addon with name: 'addon1' can not be renamed. New name: 'addon2'", exception.getMessage());
     }
 
+    @Test
+    public void shouldThrowExceptionWhenAddonConcurrencyOverwrite() {
+        initRoles();
+
+        AddonDto addonDto = createDto("1");
+        addonFacade.createAddon(addonDto);
+
+        OptimisticLockConflictException exception = Assertions.assertThrows(
+                OptimisticLockConflictException.class,
+                () -> addonFacade.updateAddon(addonDto.getName(), addonDto, "test")
+        );
+        Assertions.assertEquals("Optimistic lock conflict on update: addonName:'addon1'"
+                + ". Reload the data.", exception.getMessage());
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenHashIsNull() {
```

Comment:
pls add positive test cases with prevHash != "*"

Reply:
Done. 
A new [PR](https://github.com/epam/ai-dial-admin-backend/pull/298) has been created with these changes, targeting the development branch. The latest [commit](https://github.com/epam/ai-dial-admin-backend/pull/298/commits/c29fe9b693b8f3a4d1ce21cd4e9acd0134a4e5fe) is based on the latest issue.